addons/mail/models/discuss/discuss_channel.py

```diff
@@ -250,6 +252,39 @@ def _compute_self_member_id(self):
         for channel in self:
             channel.self_member_id = member_by_channel.get(channel)
 
+    @api.depends_context("uid", "guest")
+    @api.depends("self_member_id", "channel_member_ids", "channel_member_ids.channel_role")
+    def _compute_is_channel_owner_admin(self):
+        for channel in self:
+            # sudo: discuss.channel.member - reading channel role is accepted for self member
+            channel.is_self_channel_owner = bool(
+                channel.self_member_id and channel.self_member_id.sudo().channel_role == "owner",
+            )
+            # sudo: discuss.channel.member - reading channel role is accepted for self member
+            channel.is_self_channel_admin = bool(
+                channel.self_member_id and channel.self_member_id.sudo().channel_role == "admin",
+            )
+
+    def _search_is_channel_owner(self, operator, operand):
+        if operator != 'in':
+            return NotImplemented
+        current_partner = self.env["res.partner"]._get_current_persona()[0]
+        if current_partner:
+            channels = current_partner.channel_member_ids.filtered(lambda member: member.sudo().channel_role == "owner").channel_id
```

Comment:
Reading `channel_member_ids` doesn't scale well.
It would be better to make a specific query to find members where channel_role is the one expected.
I don't think we need to make it in 2 steps, we could directly use `channel_member_ids` with "any" in the main search no?